docs/Documenting.md

```diff
@@ -2,44 +2,57 @@
-work on if you do not want to at the time.
-```
+
+However to keep the documentation readable in text and favor consistency, the Doxygen commands that should be used are restricted to what this document mentions.
+
 To generate a doxygen manual for the project, ensure you have doxygen installed and then cd into the project root directory and run `doxygen Doxyfile`.
 
+The documentation can then be browsed by opening `docs/doxygen/html/index.html` in a web browser.
+
 ## Documenting Functions
-For functions, a description of the function's purpose should go above the function:
+
+Any comments inside functions, except bugs ([see below](#documenting-bugs)), should use `//`-style comments, even if spanning over multiple lines.
+
+A simple example of documenting a function withjust a description (note the leading `/**`):
+
 ```c
 /**
- * My description of this function
+ * Update the crawl sound timer, and play the crawling sound when it reaches 0.
  */
-void foo(void);
+void EnInsect_UpdateCrawlSfx(EnInsect* this) {
 ```
-Further considerations:
-- Any comments inside the function should follow the usual `//` or `/**/` comment styles.
-- For documenting return values, place a `@return` at the bottom of the function comment followed by the description of the return value. This should only be done if the name of the function is not self-explanatory and is well-understood.
-- For documenting parameters, place a `@param` between the comment and the @return (if applicable) followed by the name of the parameter and a brief description. This should only be done if the name of the parameter is not self-explanatory and is well-understood.
-- All `@param`s should come before `@return` and be in the same order the parameters appear in the function declaration. Note that this does not mean you should add empty `@params` for parameters deemed self-explanatory.
 
-A full example would be as follows: (however in practice functions such as this would be considered self-explanatory)
+A more complete example:
+
 ```c
 /**
- * This is an example
+ * Request to either increase or consume magic.
+ *
+ * @param amount the positive-valued amount to either increase or decrease magic by
+ * @param type how the magic is increased or consumed.
  *
- * @param bar the input
- * @return bar multiplied by 2
+ * @return false if the request failed
  */
-s32 foo(s32 bar) {
-    return 2*bar;
-}
+s32 Magic_RequestChange(PlayState* play, s16 amount, s16 type) {
 ```
 
+Note:
+
+- Documentation for self-explanatory arguments (`@param`) and return values (`@return`) may be omitted.
+- `@param` commands should not have empty lines in between.
+- Different commands (main description, `@param`, `@return`, ...) should be separated by empty lines.
+
+Other directives that may be used for documenting functions:
+
+- `@see` to reference something else ([see below](#linking-related-information)).
+- `@note` to bring attention to some of the function behavior.
```

Comment:
I think it is appropriate to have something stronger than `@note` for some things where stuff will break badly if instructions are not followed.

Reply:
Yeah and i think the default should be `@remark` since `@note` is not for trivia stuff but it would end up being / already is used this way

The problem would be that graduation is completely arbitrary though, or can we come up with actual clear "rules"?

I made this to showcase `@remark`, `@note`, `@attention`, `@warning`

<details>

```c
/**
 * Spawn an object file of a specified ID that will persist through room changes.
 *
 * This only starts loading the object, the data will not be available immediately when the function returns.
 *
 * Used for gameplay_keep, Link's object, and the subkeep if any.
 *
 * @return The new object slot corresponding to the requested object ID.
 *
 * @remark This function is not meant to be called externally to spawn object files on the fly.
 * @note When an object is spawned with this function, all objects that come before it in the entry list will be treated as
 * @attention persistent, which will likely cause the memory available for object space to run out.
 * @warning This function is only meant to be called internally on scene load, before the object list from any room is processed.
 */
```

![image](https://user-images.githubusercontent.com/5306619/172933247-2dc9af58-3edb-40ef-a257-e9cc60a7dd22.png)

</details>

Reply:
I usually use `@remark` like I'd use "Remark:" in a paper, as a tangential thing that's not usage-specific, like what the original name of the function is if known.

Reply:
I usually use `@remark` like I'd use "Remark:" in a paper, as a tangential thing that's not usage-specific, like what the original name of the function is if known.

Reply:
`@warning` is tricky because it can be mixed up with compiler warnings, but I don't love `@attention`.